src/main/drivers/pwm_output.c

```diff
@@ -387,25 +401,89 @@ void pwmWriteDshotCommand(uint8_t index, uint8_t motorCount, uint8_t command)
+        case DSHOT_CMD_BEACON5:
+            repeats = 1;
+            timeDelayUs = DSHOT_BEEP_DELAY_US;
             break;
         default:
             repeats = 1;
+            timeDelayUs = DSHOT_COMMAND_DELAY_US;
+
             break;
         }
+        if (blocking) {
+            for (; repeats; repeats--) {
+                for (uint8_t i = 0; i < motorCount; i++) {
+                    delayMicroseconds(DSHOT_COMMAND_DELAY_US);
+                    if ((i == index) || (index == ALL_MOTORS)) {
+                        motorDmaOutput_t *const motor = getMotorDmaOutput(i);
+                        motor->requestTelemetry = true;
+                        pwmWriteDshotInt(i, command);
+                    }
+                }
 
-        for (; repeats; repeats--) {
+                pwmCompleteDshotMotorUpdate(0);
+            }
+            delayMicroseconds(timeDelayUs);
+        } else {
             for (uint8_t i = 0; i < motorCount; i++) {
                 if ((i == index) || (index == ALL_MOTORS)) {
-                    motorDmaOutput_t *const motor = getMotorDmaOutput(i);
-                    motor->requestTelemetry = true;
-                    pwmWriteDshotInt(i, command);
+                    if (dshotCommandControl.command[i] == 0) {
+                        dshotCommandControl.command[i] = command;
+                        dshotCommandControl.repeats = repeats;
+                        dshotCommandControl.nextCommandAt = timeNowUs + DSHOT_COMMAND_DELAY_US;
+                        dshotCommandControl.delayAfterCommand = timeDelayUs;
+                    }
                 }
             }
+        }
+    }
+}
+
+
+bool pwmIsDshotCommand(void)
+{
+    if (dshotCommandControl.nextCommandAt) {
+        return(true);
+    } else {
+        return(false);
+    }
+}
+
+uint8_t pwmGetDshotCommand(uint8_t index)
+{
+    return(dshotCommandControl.command[index]);
+}
 
-            pwmCompleteDshotMotorUpdate(0);
-            delay(1);
+bool pwmProcessingDshotCommand(uint8_t motorCount)
+{
+    if (dshotCommandControl.nextCommandAt) {
+        timeUs_t timeNowUs = micros();
+        if (cmpTimeUs(timeNowUs,dshotCommandControl.nextCommandAt) > 0) {
+            if (dshotCommandControl.repeats > 0) {
+                dshotCommandControl.repeats--;
+                dshotCommandControl.nextCommandAt = timeNowUs + DSHOT_COMMAND_DELAY_US;
+                if (dshotCommandControl.repeats == 0) { 
+                    dshotCommandControl.nextCommandAt = timeNowUs + dshotCommandControl.delayAfterCommand;
+                }
+            } else {
+                for (uint8_t i = 0; i < motorCount; i++) {
+                        dshotCommandControl.command[i] = 0;
+                }                
+                dshotCommandControl.nextCommandAt = 0;
+                dshotCommandControl.delayAfterCommand = 0;
+            }
+        } else {
+            return(true); //Skip motor update
         }
     }
+    return(false);
```

Comment:
Coding standard for Betaflight is `return false`;

Reply:
Oops yeah, I'll fix those